### ISSUE TITLE
Tied Words 🪢 

### ISSUE THREAD

**Issue by MatthewSzurkowski** [GBADsInformatics/GBADsDataSearchUIAPI#1]
Certain words need to be tied together like "North America" and "Polar bears" in a query.

**Comment by kassyray** [comment]
Are you thinking named entity disambiguation (NED) here?

**Comment by MatthewSzurkowski** [comment]
@kassyray thats exactly it! I didn't realize that there was a name for it, thank you.

**Comment by MatthewSzurkowski** [comment]
Almost complete, trying to employ a new NLP approach on a separate branch but will update main branch once all testing is complete.

**Comment by MatthewSzurkowski** [comment]
I've created functions to try and disambiguate entities for animal species, but the general results provide harsh returns and could potentially compromise this version of the API and return incorrect results. I've continued to search for a better solution by testing different NLP libraries, which could be promising, but the best bet is to use a custom trained model which will take some time. For version 1.0, I believe we should use what we have and let the GUI iron out any issues that arise. The GUI acts as a second form of authentication which allows the API results to have some give.

For future development, I'm leaving a note here for more packages to checkout:

- https://pypi.org/project/ned/1.11.1/
- https://github.com/urchade/GLiNER

I'll also be pushing the old commit to a separate branch called **ned-requested-call**
